examples/layout/lakes/step5/test/widget_test.dart

```diff
@@ -1,15 +1,24 @@
 // Basic Flutter widget test. Learn more at https://flutter.io/docs/testing.
 
+import 'package:flutter/foundation.dart';
 import 'package:flutter_test/flutter_test.dart';
 
 import 'package:layout/main.dart';
 
 void main() {
   testWidgets('Codelab smoke test', (WidgetTester tester) async {
+    var exception;
+
+    FlutterError.onError = (FlutterErrorDetails details) async {
```

Comment:
A comment here would be helpful, since this is not "normal" -- only codelabs will want to do this.

Reply:
@filiph, how about this:

```dart
  testWidgets('Codelab smoke test', (WidgetTester tester) async {
    // We don't usually expect the Flutter framework to throw a FlutterError
    // during a smoke test but exceptionally, the app under test here does. In
    // this particular case, the error is expected because we know that the app
    // represents an intermediate step towards the development of the final
    // version.
    var exceptions = [];
    FlutterError.onError = (FlutterErrorDetails details) async {
      exceptions.add(details.exception);
    };
   ...
}
```

Reply:
Comment lgtm. It could be slightly simpler:

// A FlutterError shouldn't normally occur during a smoke test, but it
// is expected for this not-quite-finished app.